### PR TITLE
Ros2 devel

### PR DESCRIPTION
**Pull request by vaibhavbhadade**
This PR is to make basic CPP changes to support "diagnostic_updater" package under "diagnostics" ros2 Bouncy.

Here we have created new ros2_devel branch and merged changes from bponsler:ros2-devel branch. Additionally we have updated some CPP code as per ROS2 guidelines to support ROS2 Bouncy.

### CODE REVIEW

File: diagnostic_aggregator/CMakeLists.txt [hunk 1]
```diff
@@ -1,26 +1,65 @@
...
+#endif()
+#if( (${diagnostic_msgs_VERSION} VERSION_EQUAL "1.12.0") OR
+#    (${diagnostic_msgs_VERSION} VERSION_GREATER "1.12.0" AND
+#     ${diagnostic_msgs_VERSION} VERSION_LESS ${REQUIRED_diagnostic_msgs_VERSION_Jade}))
+#  message(FATAL_ERROR "diagnostic_msgs version ${REQUIRED_diagnostic_msgs_VERSION_Jade} or newer is required to build diagnotic_aggregator on #ROS Jade")
+#endif()
+
+#find_package(Boost REQUIRED COMPONENTS system)
+
+#include_directories(include ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS} gtest-1.7.0/include)
```
Comment (trainman419) [inline diagnostic_aggregator/CMakeLists.txt]: Dead code. Remove.

File: diagnostic_aggregator/CMakeLists.txt [hunk 2]
```diff
@@ -1,26 +1,65 @@
...
+find_package(diagnostic_msgs REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclpy REQUIRED)
+find_package(xmlrpcpp REQUIRED)
+
+
+#catkin_package(DEPENDS diagnostic_msgs pluginlib roscpp rospy xmlrpcpp bondcpp
+#    INCLUDE_DIRS include
+#    LIBRARIES ${PROJECT_NAME})
```
Comment (trainman419) [inline diagnostic_aggregator/CMakeLists.txt]: Dead code. Remove.

File: diagnostic_aggregator/CMakeLists.txt [hunk 3]
```diff
@@ -1,26 +1,65 @@
...
-endif()
-if( (${diagnostic_msgs_VERSION} VERSION_EQUAL "1.12.0") OR
-    (${diagnostic_msgs_VERSION} VERSION_GREATER "1.12.0" AND
-     ${diagnostic_msgs_VERSION} VERSION_LESS ${REQUIRED_diagnostic_msgs_VERSION_Jade}))
-  message(FATAL_ERROR "diagnostic_msgs version ${REQUIRED_diagnostic_msgs_VERSION_Jade} or newer is required to build diagnotic_aggregator on ROS Jade")
-endif()
-
-find_package(Boost REQUIRED COMPONENTS system)
-include_directories(include ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS} gtest-1.7.0/include)
+#find_package(catkin REQUIRED diagnostic_msgs pluginlib roscpp rospy xmlrpcpp bondcpp)
```
Comment (trainman419) [inline diagnostic_aggregator/CMakeLists.txt]: Dead code. Remove
```suggestion
```

File: diagnostic_aggregator/CMakeLists.txt [hunk 4]
```diff
@@ -1,26 +1,65 @@
-# http://ros.org/doc/groovy/api/catkin/html/user_guide/supposed.html
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
+
+#cmake_minimum_required(VERSION 2.8.3)
```
Comment (trainman419) [inline diagnostic_aggregator/CMakeLists.txt]: Dead code. Remove.
```suggestion
```

File: diagnostic_aggregator/include/diagnostic_aggregator/aggregator.h [hunk 5]
```diff
@@ -39,25 +39,34 @@
...
+//#include "diagnostic_msgs/msg/add_diagnostics.h"
+#include "diagnostic_msgs/srv/add_diagnostics.hpp"
 #include "XmlRpcValue.h"
 #include "diagnostic_aggregator/analyzer.h"
 #include "diagnostic_aggregator/analyzer_group.h"
 #include "diagnostic_aggregator/status_item.h"
 #include "diagnostic_aggregator/other_analyzer.h"
 
-
+//TODO(tfoote replace these terrible macros)
```
Comment (trainman419) [inline diagnostic_aggregator/include/diagnostic_aggregator/aggregator.h:63]: Please attribute this TODO to the correct author.

File: diagnostic_aggregator/include/diagnostic_aggregator/analyzer_group.h [hunk 6]
```diff
@@ -43,17 +43,26 @@
...
+/*#include <boost/shared_ptr.hpp>*/
+#include <memory>
 #include "XmlRpcValue.h"
 #include "diagnostic_aggregator/analyzer.h"
 #include "diagnostic_aggregator/status_item.h"
 #include "pluginlib/class_loader.hpp"
 #include "pluginlib/class_list_macros.hpp"
 
+
+//TODO(tfoote replace these terrible macros)
```
Comment (trainman419) [inline diagnostic_aggregator/include/diagnostic_aggregator/analyzer_group.h:59]: Please attribute this TODO to the correct author.

**Review by trainman419 — CHANGES_REQUESTED** [review]
It looks like there's a lot of dead code and comments in these changes that need to be cleaned up before merging.

It also looks like Travis CI is failing on these changes. It looks like the CI configuration needs to be updated to build against ROS2 instead of ROS1.

**Comment by Rohita83** [comment]
@trainman419, Thanks for your comments. As these changes are related to diagnostic_aggregator, we will fix it soon. 
Could you suggest us what changes required in CI configuration to build it against ROS2.

File: diagnostic_aggregator/CMakeLists.txt (hunk 2 above)
Comment (vaibhavbhadade) [reply diagnostic_aggregator/CMakeLists.txt]: we have fix and added this in new pull request

File: diagnostic_aggregator/CMakeLists.txt (hunk 1 above)
Comment (vaibhavbhadade) [reply diagnostic_aggregator/CMakeLists.txt]: Dead code is removed in new pull request

File: diagnostic_aggregator/CMakeLists.txt (hunk 3 above)
Comment (vaibhavbhadade) [reply diagnostic_aggregator/CMakeLists.txt]: dead code is removed in new pull request

**Review by vaibhavbhadade — COMMENTED** [review]
changes accepted

**Review by vaibhavbhadade — COMMENTED** [review]
changes accepted

**Review by vaibhavbhadade — COMMENTED** [review]
changes accepted

**Review by vaibhavbhadade — COMMENTED** [review]
changes accepted

**Review by vaibhavbhadade — COMMENTED** [review]
changes accepted

**Review by vaibhavbhadade — COMMENTED** [review]
changes accepted

**Review by vaibhavbhadade — COMMENTED** [review]
changes accepeted

**Review by vaibhavbhadade — COMMENTED** [review]
changes accepted

**Pull request closed without merging**